### PR TITLE
Frontend changes for switchable email verification feature.

### PR DESCRIPTION
**Pull request by TheDheeraj007**
Here, we are introducing a new flag disabledOrConfirmed.

disabledOrConfirmed = true, Email verification is disabled or email has already been verified. In either case, we don't show any notification and features related to the same.

disabledOrConfirmed = false, Email verification is enabled and email has not been verified. In either case, we show any notification and feature related to the same.

### CODE REVIEW

File: src/graphql/types.ts [hunk 1]
```diff
@@ -134,6 +134,7 @@ export type User = {
   githubId?: Maybe<Scalars['Int']>
   resources: Array<Resource>
   progressList: Array<Progress>
+  disabledOrConfirmed: Scalars['Boolean']
```
Comment (pbteja1998) [inline src/graphql/types.ts:137]: Did you write this out? Or auto generated using `yarn gen`?

Comment (TheDheeraj007) [reply src/graphql/types.ts:137]: I wrote it. Is this not required?

Comment (pbteja1998) [reply src/graphql/types.ts:137]: Don't write anything in that file manually. Run `yarn gen` in the terminal while keeping the backend running. That file will be updated automatically

Comment (TheDheeraj007) [reply src/graphql/types.ts:137]: ok

Comment (TheDheeraj007) [reply src/graphql/types.ts:137]: interestingly yarn gen did generate disabledOrConfirmed field here. I'll put that in the next commit.

Comment (pbteja1998) [reply src/graphql/types.ts:137]: Yeah, it will. But it's better to let it generate than for us than to take care of it individually.

Comment (TheDheeraj007) [reply src/graphql/types.ts:137]: One issue found for UI. After reloading the settings page: link and tooltip start showing up again. Not sure why though.

Comment (TheDheeraj007) [reply src/graphql/types.ts:137]: Actually when I'm having disabledOrConfirmed as true. And I'm landing on the settings page, the "resend Email verification" link, and the tool-tip is not showing up. Which is expected behavior in our case. 
However, when I reloading the same page again, they start appearing again.
I'm guessing the user object is somehow getting reset after the page reload of the settings page.

Comment (TheDheeraj007) [reply src/graphql/types.ts:137]: Never mind I'll take care of it.

Comment (TheDheeraj007) [reply src/graphql/types.ts]: Hi @pbteja1998 Hi have made the required changes can you have a look

**Pull request closed without merging**